src/sdam/topology.ts

```diff
@@ -652,40 +652,44 @@ export class Topology extends EventEmitter {
     ReadPreference.translate(options);
     const readPreference = options.readPreference || ReadPreference.primary;
 
-    this.selectServer(readPreferenceServerSelector(readPreference), options, (err, server) => {
-      if (err || !server) {
-        callback(err);
-        return;
-      }
-
-      const willRetryWrite =
-        !options.retrying &&
-        !!options.retryWrites &&
-        options.session &&
-        isRetryableWritesSupported(this) &&
-        !options.session.inTransaction() &&
-        isWriteCommand(cmd);
-
-      // increment and assign txnNumber
-      if (willRetryWrite) {
-        options.session?.incrementTransactionNumber();
-        options.willRetryWrite = willRetryWrite;
-      }
-
-      server.command(ns, cmd, options, (err, result) => {
-        if (!err) return callback(undefined, result);
-        if (!shouldRetryOperation(err)) {
-          return callback(err);
+    this.selectServer(
+      readPreferenceServerSelector(readPreference as ReadPreference),
```

Comment:
after `ReadPreference.translate` we should always be dealing with a `ReadPreference` instance, there should be no need to cast here

Reply:
There doesn't seem to be a straight forward way to type that since translate mutates the object it takes in. You think I can do something with extending the argument passed in? I tried a few ways but it can't override an incompatible type, like string and ReadPreference are not compatible.